### PR TITLE
Sparse groups and e2e coverage

### PR DESCRIPTION
**Pull request by benjaminpkane**
Adding support and graceful handling of group datasets where only the current grid slice is guaranteed to be present when navigating through samples in the modal.

### CODE REVIEW

**Comment by benjaminpkane** [comment]
Multiple media types will be a separate PR

File: e2e-pw/src/oss/fixtures/loader.ts [hunk 1]
```diff
@@ -122,14 +122,22 @@ export class OssLoader extends AbstractFiftyoneLoader {
       }
     };
 
-    if (!datasetName) {
-      await forceDatasetFromSelector();
+    if (savedView) {
```
Comment (manivoxel51) [inline e2e-pw/src/oss/fixtures/loader.ts:125]: 👍 soon we will have other params. should we append param instead

File: e2e-pw/src/oss/poms/modal.ts [hunk 2]
```diff
@@ -29,15 +41,57 @@ export class ModalPom {
...
+    return this.waitForSampleToLoad(allowErrorInfo);
+  }
+
+  async navigateSlice(
+    groupField: string,
+    slice: string,
+    allowErrorInfo = false
+  ) {
+    const currentSlice = await this.sidebarPom.getSidebarEntryText(
+      `sidebar-entry-${groupField}.name`
```
Comment (manivoxel51) [inline e2e-pw/src/oss/poms/modal.ts:66]: nit: should this be `sidebar-entry-${groupField}-name` with a dash

File: e2e-pw/src/oss/specs/groups/sparse-groups-scenes.spec.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+            )
+    dataset.add_samples(samples)
+    dataset.save_view("dynamic", dataset.group_by("dynamic", order_by="id"))`);
+});
+
+test.beforeEach(async ({ page, fiftyoneLoader }) => {
+  await fiftyoneLoader.waitUntilLoad(page, datasetName);
+});
+
+test(`ego default group slice transitions`, async ({ grid, modal }) => {
```
Comment (manivoxel51) [inline e2e-pw/src/oss/specs/groups/sparse-groups-scenes.spec.ts:56]: Very nice!

**Review by manivoxel51 — APPROVED** [review]
Nice work @benjaminpkane ! 🍨 

lots of tests 🎉

File: app/packages/state/src/recoil/groups.ts [hunk 4]
```diff
@@ -37,114 +47,125 @@ export const isGroup = selector<boolean>({
...
+      .map(({ name }) => name);
+  },
+});
+
+export const allNonPcdSlices = selector<string[]>({
+  key: "allNonPcdSlices",
+  get: ({ get }) => {
+    return get(groupMediaTypes)
+      .filter(
+        ({ mediaType }) => !["point-cloud", "point_cloud"].includes(mediaType)
```
[the remark this answers is not in the document]
Comment (benjaminpkane) [reply app/packages/state/src/recoil/groups.ts:149]: I will remove `point-cloud`. No longer valid

File: e2e-pw/src/oss/poms/modal.ts (hunk 2 above)
Comment (benjaminpkane) [reply e2e-pw/src/oss/poms/modal.ts]: Ok, we would need to follow that formatting in the react component. But that would create collisions, though this is only for testing

**Pull request merged**